cmd/bucket-notification-handlers.go

```diff
@@ -224,11 +224,17 @@ func (api objectAPIHandlers) ListenBucketNotificationHandler(w http.ResponseWrit
 		return
 	}
 
-	host, e := xnet.ParseHost(r.RemoteAddr)
-	logger.CriticalIf(ctx, e)
+	host, err := xnet.ParseHost(r.RemoteAddr)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
```

Comment:
How is this write error? We just couldn't parse the `RemoteAddr`, right?
And I think it would be a good idea to log the parsing error for tracking purposes.

Reply:
We need to write an error back to client - previously we would kill the server which is not correct @ebozduman